common/include/pcl/pcl_tests.h

```diff
@@ -219,6 +219,52 @@ namespace pcl
         return ::testing::AssertionSuccess ();
       }
 
+
+        template<typename V>
+        ::testing::AssertionResult VectorContainsAll(const char*  expr1, const char* expr2, std::vector<V> elements, std::vector<V> v) {
+            for(const V& item : elements) {
+                if(std::find(v.cbegin(), v.cend(), item)==v.cend()) {
+
+                    std::ostringstream vec_rep;
+                    std::copy(v.cbegin(), v.cend()-1, std::ostream_iterator<V>(vec_rep, ", "));
+                    vec_rep<<v.back();
+
+                    std::ostringstream elements_rep;
+                    std::copy(elements.cbegin(), elements.cend()-1, std::ostream_iterator<V>(elements_rep, ", "));
+                    elements_rep << elements.back();
+
+                    return ::testing::AssertionFailure ()
+                            << "Vector       : "<< expr2 << std::endl
+                            << "contains     : " << vec_rep.str() << std::endl
+                            << "From         : " << expr1 << std::endl
```

Comment:
I'm a bit confused how you intend to read this message. The first two lines I understand, "The actual vector ... contains ...". But what does "from" mean?

Reply:
Do you prefer this one ?
```suggestion
                             << "Actual       : " << expr2 << std::endl
                            << "contains     : " << vec_rep.str() << std::endl
                            << "Target set  : " << expr1 << std::endl
                            << "contains     : " << elements_rep.str() << std::endl
```

Reply:
Mmm I think yes, this looks more understandable.